app/helpers/authentication_helper.rb

```diff
@@ -70,4 +70,12 @@ def tooltip_text_email_or_auth_provider_btns
   def came_from_sign_up?
     request.referer&.include?(new_user_registration_path)
   end
+
+  def display_social_login?
+    return true if Authentication::Providers.enabled.include?(:apple)
+    return true if /Android/.match?(request.user_agent)
```

Comment:
This is really minor but I think it's more common to use the `String#match?` method than the `Regexp#match?` method, maybe because it feels less like a [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions) (constant part on the left instead of the right):

```suggestion
    return true if request.user_agent.match?(/Android/)
```

I think it doesn't matter much here, but it makes the next line a bit nicer since now the negation won't directly be followed by the `/`.

Reply:
Thanks for the suggestions! This indeed felt a bit off when I wrote it and TIL the term _Yoda condition_ exists. Sense it makes!

After committing the suggestions I remembered why I chose this order though: `request.user_agent` can be `nil` sometimes (it is in fact `nil` during helper tests which causes them to fail). This means using `Regexp.match?` will never fail but `String.match?` fails if the String is actually a `nil`.

Not only this, but Rubocop starts complaining about attempting to match using a plaintext Regexp (without any operators) and suggests using `.include?` and `.exclude?` with a String parameter instead. I think this makes it a bit simpler to read though, so I'm keeping it. 

The fix for the `nil` situation is adding `.to_s`. The specs make me confident everything still works as expected here.

Anyways, I'm happy to revisit this in case we want to try a different approach in a separate PR 🙂